custom/icds_reports/tests/test_aggregation_script.py

```diff
@@ -0,0 +1,484 @@
+                        row[idx] = value.encode('utf-8')
+                    elif value is None:
+                        row[idx] = 'None'
+                yield dict(zip(columns, row))
+
+    def _fasterAssertListEqual(self, list1, list2):
+        if len(list1) != len(list2):
+            self.fail('Lists are not equal')
+
+        messages = []
+
+        for idx in range(len(list1)):
+            dict1 = list1[idx]
+            dict2 = list2[idx]
+
+            differences = []
+
+            for key in dict1.keys():
+                value1 = dict1[key]
+                value2 = dict2[key]
+                if value1 != value2:
+                    differences.append(key)
+
+            if differences:
+                messages.append("""
+                    Actual and expected row {} are not the same
+                    Actual:   {}
+                    Expected: {} 
+                """.format(
+                    idx + 1,
+                    ', '.join(['{}: {}'.format(difference, dict1[difference]) for difference in differences]),
+                    ', '.join(['{}: {}'.format(difference, dict2[difference]) for difference in differences])
+                ))
+
+        if messages:
+            self.fail('\n'.join(messages))
+
+    def _load_and_compare_data(self, table_name, path, sort_key=None):
+        sort_key = sort_key or (lambda x: x)
+        self._fasterAssertListEqual(
+            sorted(
+                list(self._load_data_from_db(table_name)),
+                key=sort_key
+            ),
+            sorted(
+                self._load_csv(path),
+                key=sort_key
+            )
+        )
+
+    @classmethod
+    def setUpTestData(cls):
+        engine = connection_manager.get_session_helper('default').engine
+        metadata = sqlalchemy.MetaData(bind=engine, reflect=True)
+        path = os.path.join(os.path.dirname(__file__), 'fixtures')
+        for file_name in os.listdir(path):
+            with open(os.path.join(path, file_name)) as f:
+                table_name = FILE_NAME_TO_TABLE_MAPPING[file_name[:-4]]
+                table = metadata.tables[table_name]
+                csv_data = [row for row in csv.DictReader(f) if 'doc_id' not in row or row['doc_id']]
+                for row in csv_data:
+                    for key, value in six.iteritems(row):
+                        if value == '':
+                            row[key] = None
+                postgres_copy.copy_from(f, table, engine, format='csv', null='', header=True)
+
+                with engine.begin() as connection:
+                    connection.execute(table.insert().values(csv_data))
+
+    @classmethod
+    def setUpClass(cls):
+        super(AggregationScriptTest, cls).setUpClass()
+        connection = connections['default']
+        with connection.cursor() as cursor:
+            path = os.path.join(os.path.dirname(__file__), 'sql_templates', 'generate_tables.sql')
```

Comment:
Instead of using generate_tables.sql, these tables should be generated from the UCRs.

Look at https://github.com/dimagi/commcare-hq/blob/master/custom/icds_reports/ucr/tests/base_test.py?utf8=%E2%9C%93#L57-L67 for an example of doing that